webauthn-server-core/src/main/java/com/yubico/webauthn/OriginMatcher.java

```diff
@@ -0,0 +1,51 @@
+package com.yubico.webauthn;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+class OriginMatcher {
+
+    static boolean isAllowed(
+        String origin,
+        Set<String> allowedOrigins,
+        boolean allowPort,
+        boolean allowSubdomain
+    ) {
+        return allowedOrigins.stream().anyMatch(allowedOriginString -> {
+            if (allowedOriginString.equals(origin)) {
+                return true;
+            } else {
+                final URL allowedOrigin;
+                try {
+                    allowedOrigin = new URL(allowedOriginString);
+                } catch (MalformedURLException e) {
+                    return false;
+                }
+
+                final URL originUrl;
+                try {
+                    originUrl = new URL(origin);
+                } catch (MalformedURLException e) {
+                    return false;
```

Comment:
Same as the above

Reply:
Hmyeah. This `origin` is an input from the user-agent, so failures in this one should be somewhat expected, but I guess logging it on debug level makes sense.

Reply:
One can never have too much debug logging :-)